src/context/ShoppingCartCountContext.tsx

```diff
@@ -0,0 +1,33 @@
+import React, {
+  createContext,
+  useState,
+  useEffect,
+  PropsWithChildren,
+} from 'react';
+import { fetchShoppingCartQuantity } from '../api';
+export const CartContext = createContext<{
+  counts: number;
+  setCounts: React.Dispatch<React.SetStateAction<number>>;
+}>({
+  counts: 0,
+  setCounts: () => {},
+});
+
+export const CartProvider: React.FC<PropsWithChildren> = ({ children }) => {
+  const [counts, setCounts] = useState<number>(0);
```

Comment:
```suggestion
  const [counts, setCounts] = useState(0);
```
기본값으로 타입추론 되어서, 타입 선언이 필요 없어보여요! 
혹시 치코는 타입 선언을 해주는 경우와 타입추론을 사용하는 경우에 대해 기준이 있으신가용?

Reply:
저 같은 경우는 타입스크립트의 타입 추론에 의지하고 그것으로 인해 에러가 발생할 경우에 타입 선언을 사용하고 있습니다. 호프말대로 이번에 타입 추론이 필요없었네요! 제가 왜 저기다 number을 넣었는지... 수정했습니다.